Tests/UnitTests/Networking/Responses/OfferingsDecodingTests.swift

```diff
@@ -65,3 +104,12 @@ class OfferingsDecodingTests: BaseHTTPResponseTest {
     }
 
 }
+
+private extension Collection {
+
+    /// Returns the element at the specified index if it exists, otherwise nil.
+    subscript (safe index: Index) -> Element? {
```

Comment:
Cool this is exactly what I had in mind 👍🏻 
I can extract this an add unit tests for it in a separate PR since I think it's useful to have.